rotkehlchen/tasks/manager.py

```diff
@@ -550,59 +550,58 @@ def _maybe_update_snapshot_balances(self) -> Optional[list[gevent.Greenlet]]:
 
     def _maybe_query_produced_blocks(self) -> Optional[list[gevent.Greenlet]]:
         """Schedules the blocks production query if enough time has passed"""
+        if self.chains_aggregator.get_module('eth2') is None:
+            return None
+
+        if self.chains_aggregator.beaconchain.produced_blocks_lock.locked():
+            return None  # task is already running, either api or periodic
+
+        # Get indices for validators that haven't been queried yet,
+        # or were last queried long enough ago that they need queried again
         with self.database.conn.read_ctx() as cursor:
-            result = self.database.get_static_cache(
-                cursor=cursor, name=DBCacheStatic.LAST_PRODUCED_BLOCKS_QUERY_TS,
+            key_name = DBCacheDynamic.LAST_PRODUCED_BLOCKS_QUERY_TS.value[0].replace('{index}', '')
```

Comment:
so this is very similar to what you have in beaconchain/service.py `get_and_store_produce_blocks`. Can you abstract this out into a function?